### PR TITLE
DSD-1677: Adding interaction test examples

### PR DESCRIPTION
**Pull request by EdwinGuzman**
Fixes JIRA ticket [DSD-1677](https://jira.nypl.org/browse/DSD-1677)

## This PR does the following:

- Adds `@storybook/addon-interactions` to Storybook for interaction tests in the "With Controls" Story.
- Interaction tests should be added in the `WithControls` example in `[component].stories.tsx` files only.
- The syntax resembles React Testing Library.
- More tickets will be created from this.
- Storybook [interaction tests](https://storybook.js.org/docs/writing-tests/interaction-testing) docs.

## How has this been tested?

- Review https://nypl-design-system-git-dsd-1677-storybook-tests-nypl.vercel.app/?path=/story/components-form-elements-textinput--with-controls
- Click on the "rerun" icon to run the tests.

Screen Shot 2024-03-27 at 5 47 45 PM

## Accessibility concerns or updates

- N/A

### Checklist:

- [ ] I have updated the Storybook documentation accordingly.
- [ ] I have added relevant accessibility documentation for this pull request.
- [ ] All new and existing tests passed.

### Front End Review:

- [ ] Review the Vercel preview deployment once it is ready.

### CODE REVIEW

File: src/components/TextInput/TextInput.stories.tsx [hunk 1]
```diff
@@ -107,6 +108,22 @@ export const WithControls: Story = {
...
     jest: "TextInput.test.tsx",
   },
+  play: async ({ canvasElement }) => {
+    const textInput = within(canvasElement).getByRole("textbox");
+    await userEvent.click(textInput);
+    await userEvent.type(textInput, "Hello World");
+
+    await waitFor(() => {
+      expect(textInput).toHaveValue("Hello World");
+    });
```
Comment (bigfishdesign13) [inline src/components/TextInput/TextInput.stories.tsx:118]: In the list of interactions in Storybook, this shows as `waitFor(anonymous)`.  Is it possible to show more infomration about what is happening within the `waitFor` method?

**Comment by bigfishdesign13** [comment]
This is great.  Some thoughts...
- Can you add a link to the PR description pointing to the official Storybook documentation about interaction tests?  
- I feel we should define a basic battery of tests that should be added to all components.
- The list of interactions essentially shows straight code.  Most non-coders will not understand what is shown.  Does that matter?

File: src/components/TextInput/TextInput.stories.tsx (hunk 1 above)
Comment (EdwinGuzman) [reply src/components/TextInput/TextInput.stories.tsx:118]: I tried but could not get it to display.

**Comment by EdwinGuzman** [comment]
> Can you add a link to the PR description pointing to the official Storybook documentation about interaction tests?

Done

> I feel we should define a basic battery of tests that should be added to all components.

Yes, we can try this.

> The list of interactions essentially shows straight code. Most non-coders will not understand what is shown. Does that matter?

It's generally meant for devs to see the tests. I was hoping it would display more English-like text but haven't seen many examples with plain English.

**Review by oliviawongnyc — APPROVED** [review]
Agree with Marty's comments, wish the tests were more easily legible, but otherwise looks good to me!

File: package.json [hunk 2]
```diff
@@ -88,9 +88,10 @@
     "@storybook/addon-jest": "7.5.1",
     "@storybook/addon-links": "7.5.1",
     "@storybook/blocks": "7.5.1",
+    "@storybook/jest": "^0.2.3",
```
Comment (7emansell) [inline package.json]: nit but should this version be locked?

**Review by 7emansell — APPROVED** [review]
Lgtm, just the package version

**Pull request merged**